File: back-end/src/main/java/es/ull/project/application/service/workflow/WorkFlowDeleteService.java
package es.ull.project.application.service.workflow;

import java.util.Optional;
import java.util.UUID;

import org.springframework.beans.factory.annotation.Autowired;

import es.ull.project.application.repository.WorkFlowRepository;
import es.ull.project.application.usecase.workflow.WorkFlowDeleteUseCase;
import es.ull.project.domain.entity.WorkFlow;

public class WorkFlowDeleteService implements WorkFlowDeleteUseCase {

    @Autowired
    private WorkFlowRepository repository;

    @Override
    public Optional<WorkFlow> deleteWorkFlow(UUID id) {
        return Optional.ofNullable(repository.delete(id));
    }
}

File: front-end/src/application/service/workflow/workflow-delete-service.ts

import { WorkFlowRepository } from '@/application/repository/workflow-repository';
import { WorkFlowDeleteUseCase } from '@/application/usecase/workflow/workflow-delete-use-case';
import { WorkFlow } from '@/domain/entity/workflow';
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError, UllUUID } from '@ull-tfg/ull-tfg-typescript';

export class WorkFlowDeleteService implements WorkFlowDeleteUseCase {

  constructor(private readonly flowRepository: WorkFlowRepository) {}

  async deleteWorkFlow(flowId: UllUUID): Promise<Either<DataError, WorkFlow>> {
    return this.flowRepository.deleteWorkFlow(flowId);
  }
}
